aggregator/src/events/events_store.rs

```diff
@@ -0,0 +1,131 @@
+use crate::{
+    common::{
+        constants::{
+            EVENT_STATS_DATA, EVENT_STORE, EVENT_STORE_LABLE_KEY, PATCH_PARAM_FILED_MANAGER,
+        },
+        error,
+        utils::get_release_name,
+    },
+    Cache, EventSet,
+};
+use k8s_openapi::api::core::v1::ConfigMap;
+use kube::{
+    api::{Api, Patch, PatchParams, PostParams},
+    core::ObjectMeta,
+    Client,
+};
+use snafu::ResultExt;
+use std::{collections::BTreeMap, ops::DerefMut, time::Duration};
+use tracing::info;
+
+/// Initialize a config map for storing events.
+pub async fn initialize(namespace: &str) -> error::Result<ConfigMap> {
+    let client = Client::try_default().await.context(error::K8sClient)?;
+    let release_name = get_release_name(namespace, client.clone()).await?;
+    let api: Api<ConfigMap> = Api::namespaced(client.clone(), namespace);
+    let config_map_name = format!("{release_name}-{EVENT_STORE}");
+
+    if let Some(cm) =
+        api.get_opt(config_map_name.as_str())
+            .await
+            .context(error::GetEventStoreConfigMap {
+                name: config_map_name.clone(),
+            })?
+    {
+        info!(
+            "Config map {} for events store already exists.",
+            config_map_name.clone()
+        );
+        return Ok(cm);
+    }
+
+    info!(
+        "Creating Config map {} for events store!!!",
+        config_map_name.clone()
+    );
+    let cm = create_configmap(namespace, client.clone(), config_map_name.as_str()).await?;
+    Ok(cm)
+}
+
+/// Create a config map for storing events.
+async fn create_configmap(
+    ns: &str,
+    client: Client,
+    config_map_name: &str,
+) -> error::Result<ConfigMap> {
+    let api: Api<ConfigMap> = Api::namespaced(client.clone(), ns);
+    let mut labels = BTreeMap::new();
+    labels.insert(EVENT_STORE_LABLE_KEY.to_string(), EVENT_STORE.to_string());
+    let metadata = ObjectMeta {
+        name: Some(config_map_name.to_string()),
+        labels: Some(labels),
+        ..Default::default()
+    };
+
+    let data = init_config_map_data()?;
+    let config_map = ConfigMap {
+        data: Some(data),
+        metadata,
+        ..Default::default()
+    };
+
+    let cm = api
+        .create(&PostParams::default(), &config_map)
+        .await
+        .context(error::ServiceAccountCreate {
+            name: config_map_name,
+        })?;
+
+    // Waiting for the api-server to accept the cm
+    tokio::time::sleep(Duration::from_secs(5)).await;
```

Comment:
can we replace this with a busy wait loop with delays?

Reply:
Need your input on this . Are you talking abot this https://docs.rust-embedded.org/discovery/f3discovery/09-clocks-and-timers/busy-waiting.html?

Reply:
Something like this:
```rust
while !api_server_ready().await {
   // can add backoff as well...
   tokio::time::sleep(Duration::from_secs(1)).await;
}
```

Reply:
Config mao doesn't have any status field with which we can check id the config map is ready or not as we have done in upgrades but that was for pods. 

Refer https://kubernetes.io/docs/concepts/configuration/configmap/